### DISCUSSION TITLE
What should I learn to make ios app swift or flutter?

### DISCUSSION THREAD

**Discussion by joshboyye** [community/community#104498] [Programming Help]
### Body

I'm wondering if I should learn swift os or flutter instead 

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ghost** [comment]
If you want to develop ios apps alone learn swift for sure, but want to make a app for both android and ios go for flutter. In your case I would suggest to go for swift and after you master it learn flutter

**Comment by cnd-sw** [comment]
Learn Swift for iOS-only development with native capabilities. Choose Flutter for cross-platform (including iOS) development with a single codebase, leveraging Dart language and abstracting platform intricacies.

**Comment by Cauatn** [comment]
depends what are you aiming for, if you just want to make a app for maximize users, flutter is a choice.
Swift, in my opinion, its just if you want a job working especifically for IOS

**Comment by GradByte** [comment]
If you want to focus only on iOS apps, _**Swift**_ is a better option. I recommend learning **SwiftUI** as a **framework**. UIKit has been around for over a decade and will be replaced by SwiftUI in the next few years.

**Comment by rahulshahDEV** [comment]
It depends!

Swift: For the best possible iOS experience, deep integration with Apple features, and native performance.

Flutter: For faster development, a single codebase for iOS and Android, and beautiful custom UI.

If you're unsure, start with Flutter. It's easier to learn and you can build for both platforms. If you need the absolute best iOS performance, switch to Swift later.
